### PR TITLE
Fix type validation when string keys.

### PR DESCRIPTION
**Pull request by justinlittman**
**NOTE:  Changes to openapi.yml require updating openapi.yml for sdr-api and dor-services-app and generating models - see README.**

## Why was this change made? 🤔
Descriptive type validation wasn't working when keys were strings.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

File: spec/cocina/models/description_spec.rb [hunk 1]
```diff
@@ -82,16 +82,6 @@
     end
   end
 
-  context 'with another etd' do
```
Comment (justinlittman) [inline spec/cocina/models/description_spec.rb:85]: This wasn't adding any value.

**Review by jcoyne — APPROVED** [review]

File: spec/cocina/models/validators/description_types_validator_spec.rb [hunk 2]
```diff
@@ -124,7 +124,7 @@
     it 'raises' do
       expect do
         validate
-      end.to raise_error(Cocina::Models::ValidationError, 'Unrecognized types in description: contributor1')
+      end.to raise_error(Cocina::Models::ValidationError, 'Unrecognized types in description: contributor1 (foo)')
```
Comment (jcoyne) [inline spec/cocina/models/validators/description_types_validator_spec.rb:127]: way better message!

File: spec/cocina/models/validators/description_types_validator_spec.rb [hunk 3]
```diff
@@ -441,4 +442,28 @@
...
+            ],
+            'type' => 'foo',
+            'status' => 'primary'
+          }
+        ]
+      }
+    end
+
+    it 'is not valid' do
+      expect { validate }.to raise_error(Cocina::Models::ValidationError)
```
Comment (edsu) [inline spec/cocina/models/validators/description_types_validator_spec.rb:466]: Am I reading this right that `DescriptionTypesValidator.validate` will throw a validation exception if given attributes that have string keys, but that you can still instantiate an instance of `DescriptionTypesValidator` with attributes that have string keys?

Comment (justinlittman) [reply spec/cocina/models/validators/description_types_validator_spec.rb:466]: `DescriptionTypesValidator.validate` will properly validate if given props regardless of whether keys are strings or symbols.

**Pull request merged**